### PR TITLE
feat: Present Proof - state implementation for request sent/received

### LINKED ISSUE #1505: Present Proof - need to implement request-sent and request-received states
Need to implement the following states according to [RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0037-present-proof):
* request-sent
* request-received

### PR DESCRIPTION
**Pull request by soluchok**
The following states were implemented:
* abandoning (inbound)
* request-received  (inbound)
* request-sent (outbound)
* presentation-sent  (inbound)
* proposal-sent (inbound)

Closes #1505 
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/presentproof/states.go [hunk 1]
```diff
@@ -31,7 +32,10 @@ const (
 	stateNameProposalSent     = "proposal-sent"
 )
 
-const codeInternalError = "internal"
+const (
```
Comment (Baha-sk) [inline pkg/didcomm/protocol/presentproof/states.go]: any reason for declaring const twice?

Comment (soluchok) [reply pkg/didcomm/protocol/presentproof/states.go]: what do you mean?

Comment (DRK3) [reply pkg/didcomm/protocol/presentproof/states.go]: I think he's suggesting that you can combine the two const (...) blocks into one, unless there's a specific reason you separated the two

Comment (soluchok) [reply pkg/didcomm/protocol/presentproof/states.go]: Oh, I got it. Actually, there is no reason. Code just looks more readable.

**Pull request merged**